python/lsst/pipe/base/graph/_loadHelpers.py

```diff
@@ -98,46 +93,35 @@ class DefaultLoadHelper:
     uriObject : `~lsst.daf.butler.ButlerURI` or `io.IO` of bytes
         This is the object that will be used to retrieve the raw bytes of the
         save.
+    minimumVersion : int
+        Minimum version of a save file to load. Set to -1 to load all
+        versions. Older versions may need to be loaded, and re-saved
+        to upgrade them to the latest format before they can be used in
+        production.
```

Comment:
Currently if you load in a generation 2 graph, it will upgrade it to a generation 3 graph. This graph is perfectly executable, however some transformations do happen, mainly the nodeId gets replaced with a freshly generated UUID. This will happen each time the graph is loaded, meaning loading the graph twice will cause the same node to get different ids, which can cause all sorts of problems, including for bps. This forces a user to manually ok, a load of an old graph, and thus acknowledge the upgrade, and possibly save it out if that is their intention. I will try to make that clearer in the docs.